modules/composer-ffmpeg/src/main/java/org/opencastproject/composer/impl/EncoderEngine.java

```diff
@@ -392,4 +461,526 @@ private void handleEncoderOutput(List<File> output, String message) {
+            String aInputPad) throws EncoderException {
+      pf = profiles;
+      if (vInputPad == null && aInputPad == null)
+        throw new EncoderException("At least one of video or audio input must be specified");
+      IdBuilder idbuilder = IdBuilderFactory.newInstance().newIdBuilder();
+      int size = profiles.size();
+      // Init
+      vfilter = new ArrayList<>(java.util.Collections.nCopies(size, null));
+      afilter = new ArrayList<>(java.util.Collections.nCopies(size, null));
+      // name of output pads to map to files
+      apads = new ArrayList<>(java.util.Collections.nCopies(size, null));
+      vpads = new ArrayList<>(java.util.Collections.nCopies(size, null));
+
+      vsplit = (size > 1) ? (vInputPad + "split=" + size) : null; // number of splits
+      asplit = (size > 1) ? (aInputPad + "asplit=" + size) : null;
+      this.vInputPad = vInputPad;
+      this.aInputPad = aInputPad;
+
+      int indx = 0; // profiles
+      for (EncodingProfile profile : profiles) {
+        String cmd = "";
+        String outSuffix;
+        // generate random name as we only have one base name
+        String outFileName = params.get("out.name.base") + "_" + idbuilder.createNew().toString();
+        params.put("out.name", outFileName); // Output file name for this profile
+        try {
+          outSuffix = processParameters(profile.getSuffix(), params);
+          params.put("out.suffix", outSuffix); // Add profile suffix
+        } catch (Exception e) {
+          throw new EncoderException("Missing Encoding Profiles");
+        }
+        // substitute the output file name
+        String ffmpgCmd = profile.getExtension(CMD_SUFFIX); // Get ffmpeg command from profile
+        if (ffmpgCmd == null)
+          throw new EncoderException("Missing Encoding Profile " + profile.getIdentifier() + " ffmpeg command");
+        for (Map.Entry<String, String> e : params.entrySet()) { // replace output filenames
+          ffmpgCmd = ffmpgCmd.replace("#{" + e.getKey() + "}", e.getValue());
+        }
+        ffmpgCmd = ffmpgCmd.replace("#{space}", " ");
+        String[] arguments;
+        try {
+          arguments = CommandLineUtils.translateCommandline(ffmpgCmd);
+        } catch (Exception e) {
+          throw new EncoderException("Could not parse encoding profile command line", e);
+        }
+        List<String> cmdToken = Arrays.asList(arguments);
+        // Find and remove input and filters from ffmpeg command from the profile
+        int i = 0;
+        while (i < cmdToken.size()) {
+          String opt = cmdToken.get(i);
+          if (opt.startsWith("-vf") || opt.startsWith("-filter:v")) { // video filters
+            vfilter.set(indx, cmdToken.get(i + 1).replace("\"", "")); // store without quotes
+            i++;
+          } else if (opt.startsWith("-filter_complex") || opt.startsWith("-lavfi")) { // safer to quit now than to
+            // baffle users with strange errors later
+            i++;
```

Comment:
Never used since an exception is thrown

Reply:
The exception is thrown only if ffmpgCmd cannot be parsed and tokenized.